pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-manual-form-section/advanced-settings-section.js

```diff
@@ -0,0 +1,521 @@
+          toast({
+            type: toast.types.ERROR,
+            message: m.fpNotFoundError,
+            messageValues: {
+              lorawan_phy_version,
+              frequency_plan_id,
+              code: msg => <code>{msg}</code>,
+            },
+          })
+        } else {
+          toast({
+            type: toast.types.ERROR,
+            message: m.macSettingsError,
+            messageValues: {
+              frequency_plan_id,
+              code: msg => <code>{msg}</code>,
+            },
+          })
+        }
+      }
+    }
+    const resetMacSettings = () => {
+      setDefaultMacSettings({})
+    }
+
+    if (frequency_plan_id && lorawan_phy_version) {
+      updateMacSettings()
+    } else {
+      resetMacSettings()
+    }
+    return () => {
+      isMounted = false
+    }
+  }, [
+    frequency_plan_id,
+    lorawan_version,
+    setValues,
+    setDefaultMacSettings,
+    lorawan_phy_version,
+    setValidationContext,
+  ])
+
+  // Apply field inter-dependent modifications.
+  // Note: Values set to `undefined`` will be stripped.
+  useEffect(() => {
+    setValues(values =>
+      omitBy(
+        {
+          ...values,
+          // Map device class selector to class flags.
+          supports_class_b: isClassB,
+          supports_class_c: isClassC,
+          // Map activation mode selector to support join and multicast flag.
+          supports_join: !_skip_js_registration ? isOTAA : undefined,
+          multicast: isMulticast,
+          // Reset mac settings to defaults when toggled.
+          mac_settings: _default_ns_settings
+            ? merge({}, initialValues.mac_settings, values.mac_settings, defaultMacSettings)
+            : merge({}, initialValues.mac_settings, values.mac_settings),
+          // Strip join server address when opting to skip JS registration.
+          join_server_address: !_skip_js_registration ? jsHost : undefined,
+          // Unset default settings when required field is present (`ping_slot_periodicity`).
+          _default_ns_settings: mayChangeToDefaultSettings ? _default_ns_settings : false,
+          // Reset device class selector, if multicast and set to class A.
+          _device_class: isMulticast
+            ? values._device_class === DEVICE_CLASS_MAP.CLASS_A
+              ? ''
+              : _device_class
+            : _device_class,
+        },
+        isUndefined,
+      ),
+    )
+  }, [
+    _default_ns_settings,
+    _device_class,
+    _skip_js_registration,
+    defaultMacSettings,
+    isClassB,
+    isClassC,
+    isMulticast,
+    isOTAA,
+    mayChangeToDefaultSettings,
+    setValues,
+  ])
+
+  return (
+    <Form.CollapseSection id="advanced-settings" title={m.advancedSectionTitle}>
+      <Form.Field
+        title={sharedMessages.activationMode}
+        name="_activation_mode"
+        connectedFields={['supports_class_b', 'supports_class_c', 'multicast']}
```

Comment:
Connected fields are used to prevent such fields from being stripped when the values are cleaned.

In this case, the `_activation_mode` field actually corresponds to these three fields that it controls. These fields should only be stripped when the activation mode field is stripped (which never happens in this case).